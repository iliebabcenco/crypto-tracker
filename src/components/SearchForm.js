import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

const SearchForm = () => {
  const [filter, setFilter] = useState('');
  const dispatch = useDispatch();
  const handleFilterChange = (e) => {
    setFilter(e.target.value);
    const filter = e.target.value;
    dispatch({ type: 'CHANGE_FILTER', filter });
  };
  return (
    <div>
      <h3>Currency searching</h3>
      <input type="text" placeholder="search for..." onChange={(e) => handleFilterChange(e)} value={filter} />
    </div>
  );
};

export default SearchForm;
