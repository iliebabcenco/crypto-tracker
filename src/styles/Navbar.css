.navbar {
  width: 100%;
  color: white;
  display: flex;
  margin: 0 auto;
  padding-bottom: 1rem;
  padding-top: 1rem;
  justify-content: space-around;
  border-bottom: 1px solid #c9ab81;
  background-color: rgba(21, 30, 36, 0.8);
}