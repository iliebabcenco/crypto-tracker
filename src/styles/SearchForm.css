.form-control {
  border-radius: 30px;
}
