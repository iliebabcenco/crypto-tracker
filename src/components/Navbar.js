import SearchForm from '../containers/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
