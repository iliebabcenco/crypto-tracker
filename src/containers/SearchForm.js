import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import '../styles/SearchForm.css';

const SearchForm = () => {
  const [filter, setFilter] = useState('');
  const dispatch = useDispatch();
  const handleFilterChange = (e) => {
    setFilter(e.target.value);
    const filter = e.target.value;
    dispatch({ type: 'CHANGE_FILTER', filter });
  };
  return (
    <div className="search-form-div">
      <input type="text" className="form-control me-2" placeholder="search for..." onChange={(e) => handleFilterChange(e)} value={filter} />
    </div>
  );
};

export default SearchForm;
