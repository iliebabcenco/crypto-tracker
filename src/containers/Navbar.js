import SearchForm from '../components/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
