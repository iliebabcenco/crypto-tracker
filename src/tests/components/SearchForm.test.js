import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import SearchForm from '../../containers/SearchForm';
import store from '../../store/store';
import Navbar from '../../components/Navbar';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><SearchForm /></Provider>);
  expect(asFragment(<Provider store={store}><SearchForm /></Provider>)).toMatchSnapshot();
});

it('displays correct first child', () => {
  const { container } = render(<Provider store={store}><SearchForm /></Provider>);
  expect(container.firstChild).toMatchSnapshot();
});

it('negative check', () => {
  render(<Provider store={store}><Navbar /></Provider>);
  expect(screen.queryByText('Letâ€™s do some math!')).not.toBeInTheDocument();
});
