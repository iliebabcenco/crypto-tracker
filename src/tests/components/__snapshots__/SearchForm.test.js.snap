// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct first child 1`] = `
<div
  class="search-form-div"
>
  <input
    class="form-control me-2"
    placeholder="search for..."
    type="text"
    value=""
  />
</div>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-form-div"
  >
    <input
      class="form-control me-2"
      placeholder="search for..."
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
